### PR TITLE
Issue 44

### LINKED ISSUE #44: Support to use CryptoKey in the signer
The users need to pass their private keys in the format of PEM with the client property `scalar.dl.client.private_key_pem` to sign the Scalar DL requests.

In some web application use cases, the users may not be able to retrieve the plaintext of the private key but only the CryptoKey object.

We consider adding a new client property `scalar.dl.client.private_key_cryptokey` to pass CryptoKey. When this property is provided, load the CryptoKey directly in the signer instead.

### PR DESCRIPTION
**Pull request by supl**
This PR fixes #44

### CODE REVIEW

File: signer/WebCryptoSigner.js [hunk 1]
```diff
@@ -5,33 +5,44 @@ const jsrsasign = require('jsrsasign');
...
-      this.pkcs8 = this._PKCS1ToPKCS8(this.pkcs1);
+    let key;
+    if (this.key) {
+      key = this.key;
+    } else {
+      if (!this.pkcs8) {
+        this.pkcs8 = await this._PKCS1ToPKCS8(this.pkcs1);
+      }
+      try {
+        key = await this._key(this.pkcs8);
```
[the remark this answers is not in the document]
Comment (supl) [reply signer/WebCryptoSigner.js]: @feeblefakie 
I renamed it with `_toCryptoKeyFromPkcs8`.
It is an internal function so it doesn't matter to have a long name.
`_key` is not a good name, thank you for giving a better one!

File: signer/WebCryptoSigner.js [hunk 2]
```diff
@@ -5,33 +5,44 @@ const jsrsasign = require('jsrsasign');
...
-      this.pkcs8 = this._PKCS1ToPKCS8(this.pkcs1);
+    let key;
+    if (this.key) {
+      key = this.key;
+    } else {
+      if (!this.pkcs8) {
+        this.pkcs8 = await this._PKCS1ToPKCS8(this.pkcs1);
+      }
+      try {
+        key = await this._toCryptoKeyFromPkcs8(this.pkcs8);
```
Comment (feeblefakie) [inline signer/WebCryptoSigner.js:35]: Sorry I realized that it's a bit redundant so maybe it is better.
```suggestion
        key = await this._toCryptoKeyFrom(this.pkcs8);
```

File: signer/WebCryptoSigner.js [hunk 3]
```diff
@@ -5,33 +5,44 @@ const jsrsasign = require('jsrsasign');
...
    */
   async sign(content) {
-    if (!this.pkcs8) {
-      this.pkcs8 = this._PKCS1ToPKCS8(this.pkcs1);
+    let key;
+    if (this.key) {
+      key = this.key;
+    } else {
+      if (!this.pkcs8) {
+        this.pkcs8 = await this._PKCS1ToPKCS8(this.pkcs1);
```
Comment (feeblefakie) [inline signer/WebCryptoSigner.js:32]: Sorry again it's not directly related to this PR, but for consistency, how about this ?
```suggestion
        this.pkcs8 = await this._toPkcs8From(this.pkcs1);
```

**Comment by supl** [comment]
@feeblefakie 
Thank you (again) :D
I revised the names.
Please check again.

**Pull request merged**